unity-renderer/Assets/UIComponents/Scripts/Components/ColorPicker/ColorToggle.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+[RequireComponent(typeof(Button))]
+public class ColorToggle : MonoBehaviour
+{
+    [SerializeField] private Image colorPicker;
+    [SerializeField] private Image selectionHighlight;
+
+    private bool selectedValue;
+    private Button button;
+
+    public Color Color { get; private set; }
+
+    public bool Selected
+    {
+        get => selectedValue;
+
+        set
+        {
+            selectedValue = value;
+            selectionHighlight.enabled = selectedValue;
+        }
+    }
+
+    public event System.Action<ColorToggle> OnClicked;
+
+    private void Awake()
+    {
+        button = GetComponent<Button>();
+        button.onClick.AddListener(OnClick);
+
+        Application.quitting += () =>
+        {
+            OnClicked = null;
```

Comment:
What's the purpose?

Reply:
This class was already implemented, the only thing I did was to move it into our common UI Components library. Anyway I have reviewed it and yeah, for me this part of the code doesn't make sense neither, so I have decided to remove it for now. Thanks!